components/BottomNavigation/src/private/MDCBottomNavigationItemView.m

```diff
@@ -0,0 +1,339 @@
+    } else if (titleVisibilitySelectedNever) {
+      iconImageViewCenter = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
+    }
+    CGPoint badgeCenter =
+        CGPointMake(CGRectGetMidX(self.bounds) + CGRectGetWidth(self.iconImageView.bounds) / 2,
+                    iconImageViewCenter.y - CGRectGetMidX(self.iconImageView.bounds));
+    self.label.center =
+        CGPointMake(CGRectGetMidX(self.bounds),
+                    CGRectGetMidY(self.bounds) + CGRectGetHeight(self.bounds) * 0.25f);
+    if (animated) {
+      [UIView animateWithDuration:kMDCBottomNavigationItemViewTransitionDuration animations:^(void) {
+        self.iconImageView.center = iconImageViewCenter;
+        self.badge.center = badgeCenter;
+      }];
+    } else {
+      self.iconImageView.center = iconImageViewCenter;
+      self.badge.center = badgeCenter;
+    }
+    self.label.textAlignment = NSTextAlignmentCenter;
+  } else {
+    UIUserInterfaceLayoutDirection layoutDirection = self.mdc_effectiveUserInterfaceLayoutDirection;
+    if (layoutDirection == UIUserInterfaceLayoutDirectionLeftToRight) {
+      CGPoint iconImageViewCenter =
+          CGPointMake(CGRectGetMidX(self.bounds) - CGRectGetWidth(self.bounds) * 0.2f,
+                      CGRectGetMidY(self.bounds));
+      self.iconImageView.center = iconImageViewCenter;
+      self.label.center =
+          CGPointMake(iconImageViewCenter.x + CGRectGetWidth(self.iconImageView.bounds) +
+                      CGRectGetWidth(self.label.bounds) / 2,
+                      CGRectGetMidY(self.bounds));
+      self.badge.center =
+          CGPointMake(CGRectGetMidX(self.bounds) - CGRectGetWidth(self.bounds) * 0.2f +
+                      CGRectGetWidth(self.iconImageView.bounds) / 2,
+                      iconImageViewCenter.y - CGRectGetMidX(self.iconImageView.bounds));
+      self.label.textAlignment = NSTextAlignmentLeft;
+    } else {
+      CGPoint iconImageViewCenter =
+          CGPointMake(CGRectGetMidX(self.bounds) + CGRectGetWidth(self.bounds) * 0.2f,
+                      CGRectGetMidY(self.bounds));
+      self.iconImageView.center = iconImageViewCenter;
+      self.label.center =
+          CGPointMake(iconImageViewCenter.x - CGRectGetWidth(self.iconImageView.bounds) -
+                      CGRectGetWidth(self.label.bounds) / 2,
+                      CGRectGetMidY(self.bounds));
+      self.badge.center =
+          CGPointMake(CGRectGetMidX(self.bounds) + CGRectGetWidth(self.bounds) * 0.2f +
+                      CGRectGetWidth(self.iconImageView.bounds) / 2,
+                      iconImageViewCenter.y - CGRectGetMidX(self.iconImageView.bounds));
+      self.label.textAlignment = NSTextAlignmentRight;
+    }
+  }
+}
+
+- (CGSize)sizeForText:(NSString *)text
+                 font:(UIFont *)font
+     boundingRectSize:(CGSize)boundingRectSize {
+  CGRect rect = [text boundingRectWithSize:boundingRectSize
```

Comment:
super-nit / style-wars:
This method essentially just returns the result of another method and is only called from one place.  Do we need this to be its own method?  If so, it doesn't use self and should be a class method.

Reply:
Eliminated. Now calling boundingRectWithSize directly.